### ISSUE TITLE
average loss = nan

### ISSUE THREAD

**Issue by suryabhupa** [torch/demos#42]
When training a convolutional autoencoder without the Hessian information (in train-autoencoder.lua), it looks like the average losses are equal to nan; what exactly does the Hessian information bring about? After a while, when training the convolutional autoencoder with Hessian information, the loss turns to nan. Is this expected?
